lib/gst/player/gstplayer.c

```diff
@@ -112,6 +122,9 @@ struct _GstPlayerPrivate
   GstPlayerState app_state;
   gint buffering;
 
+  GstPlayerMediaInfo  *media_info;  /* current media info */
+  GstPlayerMediaInfo  *media_info_copy; /* last emited media info */
```

Comment:
I guess one part which I am still not clear is what will happen in this scenario:

1. we emited a GstPlayerMediaInfo object (media_info) 
2. something changes and we created a new one.
3. now I have a old one from step#1, so lets unref this one. (But old one is still being referred by app and if it tries to access members of old media_info before receiving new one may cause segfault).
4. store the new one
5. emit the signal with new one.

So to solve this I was unrefing the old one only after we successfully emited the new one. I am doing something like this:

1. whenever something changes, create a new copy.  
info = gst_player_media_info_copy (self->priv->media_info);
2. emit new one
emit_media_info_updated_signal (info)
3. if had an old one, unref that one
if (self->priv->media_info_copy)
 g_object_unref (self->priv->media_info_copy)
4. point old one to new one
self->priv->media_info_copy = info.

It possible that I am over thinking and making it unnecessary complicated. Thanks for your patience and detail review feedbacks.